govcd/api.go

```diff
@@ -59,9 +58,19 @@ func (cli *Client) NewRequest(params map[string]string, method string, reqUrl ur
 		// to the processing function.
 		payload := ""
 		if req.ContentLength > 0 {
-			buf := new(bytes.Buffer)
-			buf.ReadFrom(body)
-			payload = buf.String()
+			// We try to convert body to a *bytes.Buffer
+			var ibody interface{}
+			ibody = body
+			bbody, ok := ibody.(*bytes.Buffer)
+			// If the inner object is a bytes.Buffer, we get a safe copy of the data.
+			// If it is really just an io.Reader, we don't, as the copy would empty the reader
+			if ok {
+				payload = bbody.String()
+			} else {
+				// With this content, we'll know that the payload is not really empty, but
+				// it was unavailable due to the bidy type.
+				payload = "<N/A>"
```

Comment:
"N/A" can be interpreted as Not Available. Let's be more specific and tell the real issue. For example, "Other Type". With reflection you could even print which type it holds.

Reply:
"Other type" may confuse who is reading. I just want to say that it is not available. What about:
"<Not retrieved for logging>" ?